File: examples/python/list_topics.py
#!/usr/bin/env python3
"""
Example: List all topics in a .hlib library.

Usage:
    # First install the Python bindings:
    cd dec-hlp-python && maturin develop

    # Then run:
    python list_topics.py path/to/library.hlib
"""

import sys
from dec_hlp import Library

def main():
    if len(sys.argv) != 2:
        print(f"Usage: {sys.argv[0]} <library.hlib>", file=sys.stderr)
        sys.exit(1)

    lib = Library(sys.argv[1])

    print(f"Library: {sys.argv[1]}")
    print(f"Nodes: {lib.node_count}")
    print()

    # List all root topics
    for topic_name in lib.root_topics():
        print(topic_name)

        # Show subtopics
        for child_name in lib.children([topic_name], exact=True):
            print(f"  {child_name}")

if __name__ == "__main__":
    main()

File: examples/rust/list_topics.rs
// Example: List all topics in a .hlib library file
//
// Usage: cargo run --example list_topics -- path/to/library.hlib
//
// This example opens a compiled .hlib help library and lists all level-1
// topics with their subtopics.

fn main() {
    let args: Vec<String> = std::env::args().collect();
    if args.len() != 2 {
        eprintln!("Usage: {} <library.hlib>", args[0]);
        std::process::exit(1);
    }

    let path = std::path::Path::new(&args[1]);
    let lib = match dec_hlp::library::Library::open(path) {
        Ok(lib) => lib,
        Err(e) => {
            eprintln!("Error opening library: {}", e);
            std::process::exit(1);
        }
    };

    let header = lib.header();
    println!("Library: {}", path.display());
    println!("Nodes: {}", header.node_count);
    println!("Version: {}.{}", header.version_major, header.version_minor);
    println!();

    // List all level-1 topics and their children
    for child in lib.root().children() {
        println!("{}", child.name());

        // Show subtopics
        for sub in child.children() {
            println!("  {}", sub.name());
        }
    }
}
